### ISSUE TITLE
Help! How to find the optimal config for other node setups than the ones listed in the performance sample config folder?

### ISSUE THREAD

**Issue by tjoymeed** [NVIDIA-NeMo/Megatron-Bridge#946]
Hi all,

The below is the optimal config for 2 nodes totalling 16 H100 GPUs on Slurm HPC cluster.

Now I would like to increase to 4 nodes tatalling 32 H100 GPUs on the same Slurm HPC cluster.

Is there a quick and systematic way to find the optimal config for the config?

Thanks!

-----------------------------------------------------------------------------------

https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/scripts/performance/configs/qwen3/qwen3_30b_a3b_llm_pretrain.yaml

perf_matrix:
  h100:
    num_gpus_16:
      common:
        num_gpus_per_node: 8
        seq_length: 4096
        gbs: 512
        cuda_graphs: false
        tp: 1
        pp: 2
        cp: 1
        vp: 12
        ep: 8
        etp: 1
        fsdp: false
      bf16:
        mbs: 1

**Comment by yaoyu-33** [comment]
Hi, if you are increasing the node, usually you don't need to change any config. The DP will increase automatically (x2 in your case).

**Comment by tjoymeed** [comment]
But that might not necessarily be optimal, right?

So we need to do all the iterative loops again in order to search for the new optimal when we move to 4 nodes and 8 nodes?




Also, for the 2 nodes configuration, have you tried my configs and see if it is truly better than your posted numbers?

With no Token-Dropping.

Lets stick with this as common standard.

Thanks!

**Comment by cuichenx** [comment]
Hi @tjoymeed unless you are training with a very small global batch size, typically we expect increasing DP size (reducing gradient accumulation) to be the greatest speed up. Increasing model parallelism past a certain level will incur excessive communication costs that outweigh any computation savings. Of course, this is the general case. If you want to be sure, you will have to run experiments.

**Comment by yaoyu-33** [comment]
Closing as part of issue cleanup — this issue has been stale with no recent activity. If you still need help, please open a new issue with updated details. Thank you!

**Comment by pruprakash** [comment]
## QA RCCA Analysis

**Classification**: NOT A BUG

**Reason**: Question about finding optimal config for different node setups. Performance tuning methodology question, not a bug.

**RCCA Required**: NO - This is not a code bug requiring root cause corrective action.
